File: java/facade/PaymentMethodFactory.java
package facade;

public class PaymentMethodFactory {

    public PaymentMethod createPaymentMethodOf(String paymentMethod) {
        if (paymentMethod.equalsIgnoreCase(PaymentMethodOptions.VISA.toString())) {
            return new VisaCard();
        }
        if (paymentMethod.equalsIgnoreCase(PaymentMethodOptions.MASTER_CARD.toString())) {
            return new MasterCard();
        }
        if (paymentMethod.equalsIgnoreCase(PaymentMethodOptions.PAYPAL.toString())) {
            return new Paypal();
        }
        throw new UnsupportedOperationException("Card type is not supported");
    }
}

File: PHP/src/Structural/Facade/PaymentMethodFactory.php
<?php

namespace App\Structural\Facade;

class PaymentMethodFactory
{

    /**
     *
     * @param  string $type
     * @return PaymentMethod
     */
    public function createPaymentMethodOf(string $type): PaymentMethod
    {
        switch ($type) {
            case PaymentMethodOptions::VISA:
                return new MasterCard();
            case PaymentMethodOptions::MASTER_CARD:
                return new VisaCard();
            case PaymentMethodOptions::PAYPAL:
                return new Paypal();
        }
    }
}
